store/tikv/2pc.go

```diff
@@ -717,6 +748,10 @@ func (c *twoPhaseCommitter) execute(ctx context.Context) (err error) {
 		commitDetail.Mu.Unlock()
 	}
 	if binlogChan != nil {
+		if c.isAsyncCommit() {
+			err = errors.Errorf("async commit protocol is not compatible with TiDB Binlog by now")
+			return errors.Trace(err)
```

Comment:
If binlog is enabled, I think it is better to use the old commit logic, instead of returning an error.